Assets/Mirror/Runtime/NetworkWriterPool.cs

```diff
@@ -12,26 +14,80 @@ public void Dispose()
         }
     }
 
+    /// <summary>
+    /// Pool of NetworkWriters
+    /// <para>Use this pool instead of <see cref="NetworkWriter">NetworkWriter</see> to reduce memory allocation</para>
+    /// <para>Use <see cref="Capacity">Capacity</see> to change size of pool</para>
+    /// </summary>
     public static class NetworkWriterPool
     {
-        static readonly Stack<PooledNetworkWriter> pool = new Stack<PooledNetworkWriter>();
+        /// <summary>
+        /// Mirror usually only uses up to 4 writes in nested usings,
+        /// 100 is a good margin for edge cases when
+        /// users need a lot writers at the same time.
+        ///
+        /// <para>keep in mind, most entries of the pool will be null in most cases</para>
+        /// </summary>
+        const int PoolStartSize = 100;
 
+        /// <summary>
+        /// Size of the pool
+        /// <para>If pool is too small getting writers will causes memory allocation</para>
+        /// <para>Default value: <see cref="PoolStartSize">PoolStartSize</see> </para>
+        /// </summary>
+        public static int Capacity
+        {
+            get => pool.Length;
+            set => Array.Resize(ref pool, value);
+        }
+
+        /// <summary>
+        /// Used to reset pool after running tests
+        /// </summary>
+        internal static void ResetCapacity()
+        {
+            Array.Resize(ref pool, PoolStartSize);
+        }
+
+        static PooledNetworkWriter[] pool = new PooledNetworkWriter[PoolStartSize];
+
+        static int next = -1;
+
+        /// <summary>
+        /// Get the next writer in the pool
+        /// <para>If pool is empty, creates a new Writer</para>
+        /// </summary>
         public static PooledNetworkWriter GetWriter()
         {
-            if (pool.Count != 0)
+            if (next == -1)
```

Comment:
```suggestion
            if (next < 0)
```

Reply:
== -1 makes sense in this case, as -1 is the default